README.md

```diff
@@ -186,6 +187,31 @@ method takes one argument:
 
 2. `map` must return a value of the same Functor
 
+### Contravariant Functor
+
+1. `u.contramap(a => a)` is equivalent to `u` (identity)
+2. `u.contramap(x => f(g(x)))` is equivalent to `u.contramap(f).contramap(g)`
+(composition)
+
+#### `contramap` method
+
+```hs
+contramap :: Contravariant f => f a ~> (b -> a) -> f b
```

Comment:
Sure, you can have something like.

```haskell
data Foo a = Foo
```

And that can be contravariant:

```haskell
instance Contravariant Foo where
  contramap _ _ = Foo
```

Interestingly, it's also covariant :smile: 

```haskell
instance Functor Foo where
  fmap _ _ = Foo
```

Or if you're lazy:

```haskell
{-# LANGUAGE DeriveFunctor #-}

data Foo a = Foo deriving (Functor)
```

We shouldn't use the signatures above because they do limit the spec too much. For instance `Foo` couldn't be `Contravariant` with either of those signatures.